js/ai/src/generate.ts

```diff
@@ -610,7 +610,11 @@ export async function generate<
 
   return await runWithStreamingCallback(
     resolvedOptions.streamingCallback,
-    async () => new GenerateResponse<O>(await generateAction(params))
+    async () =>
+      new GenerateResponse<O>(
```

Comment:
Do you mind adding a quick unit test for this?

Reply:
Done.